### PR TITLE
NAS-103131 / 11.3 / Add validation for dyndns service

### PR DESCRIPTION
**Pull request by sonicaj**
This commit adds validation for dyndns service

### CODE REVIEW

File: src/middlewared/middlewared/plugins/dyndns.py [hunk 1]
```diff
@@ -55,6 +55,23 @@ class Config:
...
+    @private
+    async def validate_data(self, data, schema):
+        verrors = ValidationErrors()
+        if data.get('provider') not in await self.provider_choices():
+            verrors.add(
+                f'{schema}.provider',
+                'Please select a valid provider.'
+            )
+
+        if not data.get('domain'):
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/dyndns.py]: Why can't you use empty=False on the List schema?

Comment (sonicaj) [reply src/middlewared/middlewared/plugins/dyndns.py]: My bad, i overlooked `empty` option for list schema. Thank you for letting me know

File: src/middlewared/middlewared/plugins/dyndns.py [hunk 2]
```diff
@@ -55,6 +55,17 @@ class Config:
             'ipv6tb@he.net': 'he.net'
         }
 
+    @private
+    async def validate_data(self, data, schema):
+        verrors = ValidationErrors()
+        if data.get('provider') not in await self.provider_choices():
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/dyndns.py]: Thank you, @william-gr, my bad, the provider `custom` also should be accepted; also, why `.get`?

File: src/middlewared/middlewared/plugins/dyndns.py [hunk 3]
```diff
@@ -55,6 +55,25 @@ class Config:
...
+    @private
+    async def validate_data(self, data, schema):
+        verrors = ValidationErrors()
+        provider = data['provider']
+        if provider == 'custom':
+            for k in ('custom_ddns_server', 'custom_ddns_path'):
+                if not data[k]:
+                    verrors.add(
+                        f'{schema}.{k}',
+                        'Required when using custom provider.'
```
Comment (Mrt134) [inline src/middlewared/middlewared/plugins/dyndns.py]:
```suggestion
                        'Required when using a custom provider.'
```

**Pull request merged**